### ISSUE TITLE
Multiselect to remove events

### ISSUE THREAD

**Issue by Homas** [blakeblackshear/frigate#2063]
**Describe what you are trying to accomplish and why in non technical terms**
I want to be able to select multiple events so that I can remove them at the same moment.

**Describe the solution you'd like**
I want to keep the events for long period of time 90+ days and I want to be able to remove events which I'm not interested in (e.g. false positives, detections if someone from my family is on camera, e.g. working on backyard etc)

**Describe alternatives you've considered**
You can manually delete them one by one but literally it takes time.
I also considering to be able to turn off detections for some time e.g. 30 minutes or till sunset but it will require automation and dependencies on other software e.g. hass

**Additional context**
It should be an easy fix :)

**Comment by mgf909** [comment]
2nd this request. I have a test camera in my home office, and its recorded 100's of events of me just moving a little... would love to delete them all.
Or is there a command that could clear all recordings and events for a particular camera?

**Comment by richardkuw** [comment]
3rd this request! I have a new Frigate setup and I am regularly using up all my storage while I tune my object detection. I would prefer not to have to remove all events every time, or buy more storage for spurious events.

Once the multi-select mechanism is in place it would be fantastic if selected events could be deleted OR archived.

**Comment by Homas** [comment]
I'm not an expert in snowpack and preact so I made a "dirty hack" and I'm not going to add that as a pull request. I hope someone will be able to make it in a better way. The changes are minimal so I hope it may make to the next official build.

I've added the following features:
- "has clip" filter;
- refresh page button (I didn't find a way to refresh just the table);
- (de)select all visible events;
- select events;
- delete selected events;

Known issues:
- I didn't add a new API call, so events are deleted one by one and there is no progress indicator. Depending on # of events it takes some time.
- I didn't create a nice confirmation window and used a standard javascript dialog to confirm deletion.
- has clip filter filter doesn't display correct value when applied.
- refresh button, has a wrong icon. I just reused an existing icon. 
- the refresh button refreshes the whole page. It would be better to refresh just the table.
- if all events were selected/deleted. Older events may not be loaded so you will need to refresh the page.
<img width="1677" alt="Screenshot 2022-01-02 at 10 21 25" src="https://user-images.githubusercontent.com/13776103/147885525-f768ec20-537b-4fd4-87d1-a684d3d81e07.png">

https://github.com/Homas/frigate/commit/9ff4866f44d6670ad145104265693fc14c9c1f84

**Comment by SteveALee** [comment]
Definitely need this! Is made worse by fact you cannot locally develop (at least not for registry despatch). 

You are forced into a painfully slow edit, commit, push, trigger cycle which is slow enough, and then you have to delete all those test runs (and squash your commit history)

**Comment by blakeblackshear** [comment]
> You are forced into a painfully slow edit, commit, push, trigger cycle which is slow enough, and then you have to delete all those test runs (and squash your commit history)

What are you talking about?

**Comment by MrAlucardDante** [comment]
duplicate of #2223 
can you please make it happen ?

**Comment by brilthor** [comment]
> duplicate of #2223 can you please make it happen ?

#2223 is not a duplicate of this

#2223 : Nuke all events button
this one: Ability to select multiple events in the page and delete them in one go

although this properly implemented might address the needs of #2223

**Comment by Nocturne2k** [comment]
Any chance this feature will make in in 0.11 ?

**Comment by NickM-27** [comment]
> Any chance this feature will make in in 0.11 ?

Have you used 0.11? The deleting experience is much improved as it only takes one click to delete an event.

What you see now for delete behavior in 0.11 is what will be there in final release.

**Comment by Nocturne2k** [comment]
> Have you used 0.11? The deleting experience is much improved as it only takes one click to delete an event.
> What you see now for delete behavior in 0.11 is what will be there in final release.

But there is already the possibility to select a day or a week. Sometime i just want to delete a whole day because i did testing for example. So why not add a button for "Delete all shown". So you can Filter by Day, Week, whatever and delete all events from that timeframe.

**Comment by NickM-27** [comment]
> But there is already the possibility to select a day or a week. Sometime i just want to delete a whole day because i did testing for example. So why not add a button for "Delete all shown". So you can Filter by Day, Week, whatever and delete all events from that timeframe.

I wasn't suggesting that what is in 0.11 is the optimal solution or even considered a proper solution to this issue, just that it is much improved and much faster than 0.10.

In any case, what is in 0.11 is what will be there in the final release

**Comment by SHU-red** [comment]
Just as sidenote:

Who is looking for a "emergency solution" to be able to delete all events, i just used `xdotool` to quick'n'dirty automate a mouse click every second to automatically push the delete button of frigate-events (always sitting at the same spot)

```
#!/bin/bash
while [ 1 ]; do
  xdotool click 1 &
  sleep 1
done
```

Funny to watch and saved me, although i know that this is not a "professional soution" :sweat_smile: 

Looking forward to be able to e.g. do this in frigate-gui like:
- delete all events
- delete events older than
- delete all events but exclude starred

**Comment by rgutialo** [comment]
Is there any news about this functionality?. I think this one is required for this product. Many features but this one is missing....

**Comment by gtxaspec** [comment]
+1

**Comment by AMD-NICK** [comment]
+1 

Also, I noticed that if you try to delete a starred event, the confirmation window cannot be closed without agreeing. You have to refresh the page.

**Comment by Lockie85** [comment]
+1

**Comment by dCypherNx** [comment]
+1

**Comment by iayanpahwa** [comment]
+1

**Comment by superdong69** [comment]
Sorry @NickM-27 for duplicating this request! Adding my request here, in addition, not sure if this would be a different request, but adding this to the Frigate+ Images interface would be the cats b-hole as well. ;)

**Comment by Lockie85** [comment]
@AMD-NICK why did you downvote the +1 above your post then +1 yourself?

Also why are people downvoting +1s (upvotes). Seem odd.

**Comment by AMD-NICK** [comment]
> @AMD-NICK why did you downvote the +1 above your post then +1 yourself?
> 
> Also why are people downvoting +1s (upvotes) in general. Seems odd.

idk, I think it was a miss click. Fixed

**Comment by sochartgit** [comment]
Bump, a solution please.

**Comment by Tintef** [comment]
I know it's the solution we want, but I put together a little script to loop through the events on simulate a click to delete them:

```js
const bulkDelete = async (max = Infinity, timeout = 250) => {
    try {
        const getEvent = async () => {
            await new Promise(r => setTimeout(r, timeout));
    
            const events = document.getElementsByClassName('flex bg-slate-100 dark:bg-slate-800 rounded cursor-pointer min-w-[330px]');
    
            if (!events) return undefined;
            if (events.length === 0) return undefined;
            return events[0];
        }
      
    
        let event = getEvent();
        let i = 1;
        while (event && i <= max) {
            console.log(`deleting event number ${i}`);
            event.childNodes[1].childNodes[2].childNodes[0].dispatchEvent(new Event('click'));
            event = await getEvent();
            i = i + 1;
        }
    } catch (error) {
        console.log('error: ', error);
    }
}
```

And then call it like:

```js
await bulkDelete() // note that you can setup a limit of events to delete and the ~time between clicks
```

**Comment by moogoos** [comment]
bump. have hundreds id like to delete and have to do it one by one.

**Comment by tclemos** [comment]
+1

**Comment by hapklaar** [comment]
Must be one of the most requested features and seemingly not that complex. Curious why this hasn't been implemented yet!

fwiw +1

**Comment by NickM-27** [comment]
The UI is currently being rewritten for 0.14 and multiselect to remove has been implemented, closing https://github.com/blakeblackshear/frigate/pull/10161

**Comment by mattogodoy** [comment]
In the meantime, I just created a script to do this. It's not fast, but it works:

https://github.com/blakeblackshear/frigate/assets/1627178/7a981947-d490-4ba2-9dab-a8c155f05acd

You can fins this script and the instructions [here](https://gist.github.com/mattogodoy/1618f07e0966d846bfa64bf847535293).
Hope you find it useful!

**Comment by matheusfbraga** [comment]
I was looking for this, made simple trick:

```
#!/bin/bash

events=$(curl -s http://localhost:5000/api/events)

echo $events | jq -c '.[]' | while read -r event; do
    id=$(echo $event | jq -r '.id')
    response=$(curl -s -o /dev/null -w "%{http_code}" -X DELETE "http://localhost:5000/api/events/$id")
    if [ "$response" -eq 200 ]; then
        echo "Deleted event with ID: $id"
    else
        echo "Failed to delete event with ID: $id, Status: $response"
    fi
done
```

**Comment by bobmarley2021** [comment]
+1 This is so unbelievably annoying. Frigate is an amazing project. Why is the UI missing _**really basic features?!**_

**Comment by whitefox50** [comment]
+1 here also. I recently deleted all my footage on my storage drive and now I am left with "Clips" that I can't delete except if I do it one-by-one. Please add the much deserved request, it's a very common complaint!

**Comment by NickM-27** [comment]
I'm not sure why y'all are still commenting when, as it says in my comment above (and at the top) this has already been added for the next version.

**Comment by whitefox50** [comment]
My apologies, I look forward to the nest release. Thank you for all the hard work you do. 🙏

**Comment by Lockie85** [comment]
> I'm not sure why y'all are still commenting when, as it says in my comment above (and at the top) this has already been added for the next version.

I think people are just keen (but thankful) to see the feature but generally don't want to switch to a beta build and have no understanding of when the next version will be released. I assume it's close now to that point?

**Comment by marc4s** [comment]
Sorry I still do not know how to multiselect with the new v14 beta versions.

I found that I can select them one-by-one and then delete them all. and I read the changelog from today that crtl-A multiselect was added NOW with beta4.
but before and without ctrl-A - how can we multiselect a big bunch of videos?

**Comment by marc4s** [comment]
I think I was a bit unclear.
I am searching for the "select all" button.

by the way. the date filter cannot multiselect dates/periods.
